evmcore/tx_pool_test.go

```diff
@@ -227,6 +228,9 @@ func validateTxPoolInternals(pool *TxPool) error {
 func validateEvents(events chan NewTxsNotify, count int) error {
 	var received []*types.Transaction
 
+	// add a timer to detect non-firing events
+	time.Sleep(50 * time.Millisecond)
```

Comment:
not sure I understand this ... the following loop seems to block on incoming events and has a 5-second timeout. This sleep step seems to be unnecessary. Maybe the time limit in line 249 should be raised instead?

Reply:
The problem here is that without a sleep, the check will happen right away after calling the function.
If there is a background routine that should not trigger an event, but it does. The test will validate the event before the faulty behavior happens. This complete test is a little weak in this regard. But I think that it does not worth not to rewrite it now.